### PR TITLE
Mcs 762 status info

### PR DESCRIPTION
**Pull request by cedorman**
Print out the status during a run, showing what all the status values are for the different scenes.

### CODE REVIEW

File: pipeline_ray.py [hunk 1]
```diff
@@ -183,11 +184,20 @@ def setup_logging(log_file):
 
 # Classes to keep track of status of individual scenes and (possibly
 # multiple) runs of those scenes
+class StatusEnum(Enum):
+    UNKNOWN = 0,
+    PENDING = 1,
+    RETRYING = 2,
+    SUCCESS = 3,
+    ERROR = 4,
+    ERROR_TIMEOUT = 5
```
Comment (kdrumm) [inline pipeline_ray.py:193]: I definitely like the change the enums.  I'm somewhat concerned about the ERROR_TIMEOUT value.  I think it'll work in the short term as we don't have ways to handle many different errors.

File: pipeline_ray.py [hunk 2]
```diff
@@ -344,19 +348,33 @@ def run_scenes(self):
...
+        """ During the run, print out the number of completed jobs,
+        number current running, number to go, number failed, etc """
+        logging.info(f"Status for {len(self.scene_statuses)} scenes: ")
+        frequency = {}
+        current_statuses = [scene_status.status for scene_status
+                            in self.scene_statuses.values()]
+        for scene_status in current_statuses:
+            frequency[scene_status] = current_statuses.count(scene_status)
+        for key, value in frequency.items():
+            logging.info(f"    {key} -> {value}")
```
Comment (kdrumm) [inline pipeline_ray.py:377]: We should also print this status at the very end when finished.

File: pipeline_ray.py (hunk 1 above)
Comment (cedorman) [reply pipeline_ray.py:193]: We can add more types of errors.  Or we could try to categorize them into groups, such as ERROR and ERROR_RETRYABLE.  TIMEOUT would be a retryable.

Comment (ThomasSchellenbergNextCentury) [inline pipeline_ray.py:193]: If the specific numbers don't matter, you can use `auto` here instead: https://docs.python.org/3/library/enum.html#enum.auto

Assigning specific values to Enum members implies to me that those specific values matter, but I don't think that's the case here. On the other hand, maybe I'm overthinking it :)

File: pipeline_ray.py [hunk 3]
```diff
@@ -380,9 +398,10 @@ def print_results(self):
             if file not in scenes_printed:
                 scenes_printed.append(file)
                 self.print_scene_status(s_status, "  ")
+        self.print_status()
 
     def print_scene_status(self, s_status, prefix=""):
-        logging.info(f"{prefix}Scene: {s_status.status} - '+"
+        logging.info(f"{prefix}Scene: {s_status.status} - " +
                      f"'{s_status.scene_file}")
```
Comment (ThomasSchellenbergNextCentury) [inline pipeline_ray.py:405]: You technically don't need the `+` to concatenate adjacent f-strings: https://www.python.org/dev/peps/pep-0498/#concatenating-strings

File: pipeline_ray.py [hunk 4]
```diff
@@ -431,6 +450,10 @@ def parse_args():
 if __name__ == '__main__':
     args = parse_args()
 
+    start_time = time.time()
+    date_str = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))
```
Comment (ThomasSchellenbergNextCentury) [inline pipeline_ray.py:454]: Not a big deal, but should we create a short function to generate the date string (or at least add a global variable for the format), so we know we're formatting the date consistently across the application?

File: pipeline_ray.py (hunk 1 above)
Comment (cedorman) [reply pipeline_ray.py:193]: I hadn't noticed that.  I'll use auto().   Though the syntax is really ugly.

File: pipeline_ray.py (hunk 2 above)
Comment (ThomasSchellenbergNextCentury) [inline pipeline_ray.py:377]: Using `{key}` will print the status as something like `StatusEnum.SUCCESS`, right? If you'd prefer, you can use `{key.name}` to just print `SUCCESS`

File: pipeline_ray.py (hunk 4 above)
Comment (cedorman) [reply pipeline_ray.py:454]: Done

File: pipeline_ray.py (hunk 2 above)
Comment (cedorman) [reply pipeline_ray.py]: sure, pretty easy.  Added.

File: pipeline_ray.py (hunk 3 above)
Comment (cedorman) [reply pipeline_ray.py:405]: Nice, didn't know that either.

**Comment by rartiss55** [comment]
LGTM - only thing I noticed is we may want to revisit printing each individual status at the end for long runs (maybe specific files that weren't successful would be better).

**Review by rartiss55 — APPROVED** [review]

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Pull request merged**